cmake/thirdparty/SetupHIP.cmake

```diff
@@ -36,3 +36,26 @@ blt_import_library(NAME          blt::hip_runtime
                    DEPENDS_ON    hip::host
                    TREAT_INCLUDES_AS_SYSTEM ON
                    EXPORTABLE    ${BLT_EXPORT_THIRDPARTY})
+
+if(DEFINED AMDGPU_TARGETS)
+    # If we haven't selected a particular architecture via CMAKE_HIP_ARCHITECTURES,
+    # we want to remove the unconditionally added compile/link flags from the hip::device target.
+    # FIXME: This may cause problems for targets whose HIP_ARCHITECTURES property differs
+    # from CMAKE_HIP_ARCHITECTURES
```

Comment:
>  FIXME: This may cause problems for targets whose HIP_ARCHITECTURES property differs
    from CMAKE_HIP_ARCHITECTURES

Could you please expand on this concern? Where is `HIP_ARCHITECTURES` defined?

Reply:
Found it: https://cmake.org/cmake/help/latest/prop_tgt/HIP_ARCHITECTURES.html

From the docs, it sounds like a user would have to modify the `HIP_ARCHITECTURES` property after it is initialized from the `CMAKE_HIP_ARCHITECTURES` variable.